Tests/Functional/Action/Tool/Message/MessageActionTest.php

```diff
@@ -181,6 +183,68 @@ public function testItReturnsUnauthorizedResponseWithEmptyIdToken(): void
         );
     }
 
+    public function testItRedirectOnErrorToLaunchPresentationReturnUrlWhenSpecifiedInMessageClaims(): void
+    {
+        Carbon::setTestNow(Carbon::now()->subSeconds(LtiMessageInterface::TTL + 1));
+
+        $builder = static::$container->get(LtiLaunchRequestBuilder::class);
+
+        $launchRequest = $builder->buildResourceLinkLtiLaunchRequest(
+            new ResourceLink('resourceLinkIdentifier'),
+            $this->registration,
+            null,
+            [],
+            [
+                new LaunchPresentationClaim(null, null, null, 'http://platform.com/error')
+            ]
+        );
+
+        Carbon::setTestNow();
+
+        $this->client->request(
+            Request::METHOD_GET,
+            sprintf('/test/message?%s', http_build_query($launchRequest->getParameters()))
+        );
+
+        $response = $this->client->getResponse();
+        $this->assertInstanceOf(RedirectResponse::class, $response);
+        $this->assertEquals(
+            'http://platform.com/error?lti_errormsg=LTI+message+request+authentication+failed%3A+JWT+id_token+is+expired',
+            $response->getTargetUrl()
+        );
+    }
+
+    public function testItRedirectOnErrorToLaunchPresentationReturnUrlWithQueryParamsWhenSpecifiedInMessageClaims(): void
```

Comment:
we can do a bit longer: it's `itRedirects` :smiley:

Reply:
ok, fixed